frontend/src/metabase/ui/components/overlays/Popover/use-popover.ts

```diff
@@ -111,7 +111,7 @@ export function usePopover(options: UsePopoverOptions) {
 
   return {
     floating,
-    controlled: options.opened !== null,
+    controlled: typeof options.opened === "boolean",
```

Comment:
Modify this because we overrode this line and this makes the popover's uncontrolled version not working. See a [Slack discussion](https://metaboat.slack.com/archives/C057WD5L0JG/p1703258329978889) about this.